<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>561</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Levi Fractals</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
    <item>
     <widget class="FractalView" name="fractalView" native="true"/>
    </item>
    <item>
     <layout class="QVBoxLayout" name="rightPanel">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="initialWidthLabel">
        <property name="text">
         <string>Initial width</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="widthSlider">
        <property name="minimum">
         <number>5</number>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="invertedAppearance">
         <bool>false</bool>
        </property>
        <property name="tickPosition">
         <enum>QSlider::TicksAbove</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="fractalTypeGroupBox">
        <property name="title">
         <string>Fractal type</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QRadioButton" name="radioDefault">
           <property name="text">
            <string>Default</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioPi">
           <property name="text">
            <string>Pi-shaped</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioIsland">
           <property name="text">
            <string>Island</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="animationCheckbox">
        <property name="text">
         <string>Animation</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="scaleSliderLabel">
        <property name="text">
         <string>View scale</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="scaleSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>500</number>
        </property>
        <property name="value">
         <number>200</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="buttonsSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="nextIterButton">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Next iteration</string>
        </property>
        <property name="autoDefault">
         <bool>false</bool>
        </property>
        <property name="default">
         <bool>false</bool>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>561</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>FractalView</class>
   <extends>QWidget</extends>
   <header>fractalview.h</header>
   <container>1</container>
   <slots>
    <slot>nextIter()</slot>
    <slot>setInitialWidth(int)</slot>
    <slot>toggleAnimation(bool)</slot>
    <slot>setSquareWidth(int)</slot>
    <slot>resetState()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>nextIterButton</sender>
   <signal>clicked()</signal>
   <receiver>fractalView</receiver>
   <slot>nextIter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widthSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>fractalView</receiver>
   <slot>setInitialWidth(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>429</x>
     <y>128</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioDefault</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFractalType(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>138</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioIsland</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFractalType(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>194</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioPi</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>setFractalType(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scaleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>fractalView</receiver>
   <slot>setSquareWidth(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>484</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>animationCheckbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>fractalView</receiver>
   <slot>toggleAnimation(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>fractalView</receiver>
   <slot>resetState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>535</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setFractalType(bool)</slot>
 </slots>
</ui>
